test/test_operators.py

```diff
@@ -1,74 +1,137 @@
 import unittest
-from test.util import TestCase
-
+from test.util import TVMTest
+from hypothesis import given, settings, strategies as st
 import torch
 import torch.nn.functional as F
 import torch
 
 # test jit tvm operators
-class TestOperators(TestCase):
+class TestOperators(TVMTest):
+    @TVMTest.given(shape=TVMTest.rand_shape(rank=1))
+    def test_add(self, shape):
+        x = torch.rand(shape)
+        y = torch.rand(shape)
+        z = torch.rand(shape)
 
-    def test_add(self):
         def add(a, b, c):
-          return a + b + c
+            return a + b + c
+
+        ref_out, tvm_out = self.runBoth(add, x, y, z)
+        assert torch.allclose(ref_out, tvm_out)
 
-        self.checkTraceTVM(add, verbose=True)
+    @TVMTest.given(shape=TVMTest.rand_shape(rank=1))
+    def test_mul(self, shape):
+        x = torch.rand(shape)
+        y = torch.rand(shape)
+        z = torch.rand(shape)
 
-    def test_mul(self):
         def mul(a, b, c):
             return a * b * c
 
-        self.checkTraceTVM(mul, verbose=True)
-
-    def test_conv(self):
-        def conv(a, b, c):
-            return F.conv2d(a, b) + c
-
-        def conv_bias(a, b, c, d):
-            return F.conv2d(a, b, c) + d
-
-        def conv_stride_padding_dilation(a, b, c):
-            return F.conv2d(a, b, stride=(2, 1), padding=(4, 2), dilation=(3,1)) + c
+        ref_out, tvm_out = self.runBoth(mul, x, y, z)
+        assert torch.allclose(ref_out, tvm_out)
+
+    @TVMTest.given(
+        shape=TVMTest.rand_shape(rank=4, min_dim=4, max_dim=4),
+        kernel_size=TVMTest.rand_int(3, 3),
+        num_kernels=TVMTest.rand_int(5, 5),
+    )
+    def test_conv_simple(self, shape, kernel_size, num_kernels):
+        # NCHW
+        X = torch.rand(shape)
+        W = torch.rand((num_kernels, shape[1], kernel_size, kernel_size))
+        bias = torch.rand(num_kernels)
+
+        def conv(a, b):
+            return F.conv2d(a + a, b)
+
+        def conv_bias(a, b, c):
+            return F.conv2d(a + a, b, c)
+
+        ref_out, tvm_out = self.runBoth(conv, X, W)
+        assert torch.allclose(ref_out, tvm_out, rtol=0.01, atol=0.01)
```

Comment:
I think it would be better to standardize the rtol and atol instead of calling them each time, probably make a wrapper with `torch.allclose` and make `rtol` and `atol` be defaults.

Reply:
it's not a huge amount to copy and paste over. I like avoiding wrappers in test cases so that anyone new to the project can copy paste and immediately tweak what they need